src/GovUk.Education.ExploreEducationStatistics.Notifier.Tests/NotifierFunctionsIntegrationTest.cs

```diff
@@ -1,17 +1,98 @@
+using System;
+using System.Collections.Generic;
 using System.Threading.Tasks;
+using GovUk.Education.ExploreEducationStatistics.Common.Services;
 using GovUk.Education.ExploreEducationStatistics.Common.Tests;
+using GovUk.Education.ExploreEducationStatistics.Common.Tests.Extensions;
+using GovUk.Education.ExploreEducationStatistics.Common.Tests.Utils;
+using GovUk.Education.ExploreEducationStatistics.Notifier.Configuration;
+using GovUk.Education.ExploreEducationStatistics.Notifier.Functions;
+using GovUk.Education.ExploreEducationStatistics.Notifier.Model;
+using GovUk.Education.ExploreEducationStatistics.Notifier.Services.Interfaces;
 using Microsoft.Azure.Cosmos.Table;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Options;
+using Moq;
 using Testcontainers.Azurite;
 using Xunit;
 
 namespace GovUk.Education.ExploreEducationStatistics.Notifier.Tests;
 
 public abstract class NotifierFunctionsIntegrationTest
-    (FunctionsIntegrationTestFixture fixture) : FunctionsIntegrationTest<NotifierFunctionsIntegrationTestFixture>(fixture);
+    (NotifierFunctionsIntegrationTestFixture fixture) : FunctionsIntegrationTest<NotifierFunctionsIntegrationTestFixture>(fixture), IAsyncLifetime
+{
+    public Task InitializeAsync()
+    {
+        return Task.CompletedTask;
+    }
+
+    public Task DisposeAsync()
+    {
+        MockUtils.VerifyAllMocks(fixture._emailService);
+        fixture._emailService.Reset();
+
+        return ClearAzureDataTableTestData(TableStorageConnectionString());
+    }
+
+    public string TableStorageConnectionString()
+    {
+        return fixture.TableStorageConnectionString();
+    }
+
+    public AppSettingsOptions GetAppSettingsOptions()
+    {
+        return GetRequiredService<IOptions<AppSettingsOptions>>().Value;
+    }
+
+    public GovUkNotifyOptions GetGovUkNotifyOptions()
+    {
+        return GetRequiredService<IOptions<GovUkNotifyOptions>>().Value;
+    }
+
+    public async Task AddTestSubscription(string tableName, SubscriptionEntity subscription)
+    {
+        var storageAccount = CloudStorageAccount.Parse(TableStorageConnectionString());
+        var tableClient = storageAccount.CreateCloudTableClient();
+        var table = tableClient.GetTableReference(tableName);
+        await table.CreateIfNotExistsAsync();
+
+        await table.ExecuteAsync(TableOperation.InsertOrReplace(subscription));
+    }
```

Comment:
Just wondering if using the `DataTableStorageService` to add this would work here, or is the interface on `SubscriptionEntity` different and preventing this?

Reply:
I just left this bit as-is from the previous code - this method adds the **publication** subscriptions, and all of that code is still using the deprecated Cosmos library, including this method. So I thought best to leave all of the publication related stuff as untouched as possible for now, until we come to swap everything out to use the new stuff (if/when we decide that the new stuff is the way to go!)